Source/SwiftLintFramework/Rules/MissingDocsRule.swift

```diff
@@ -0,0 +1,115 @@
+//
+//  MissingDocsRule.swift
+//  SwiftLint
+//
+//  Created by JP Simard on 11/15/15.
+//  Copyright © 2015 Realm. All rights reserved.
+//
+
+import SourceKittenFramework
+
+internal extension File {
+    func missingDocOffsets(in dictionary: [String: SourceKitRepresentable],
+                           acls: [AccessControlLevel]) -> [(Int, AccessControlLevel)] {
+        if dictionary.enclosedSwiftAttributes.contains(.override) ||
+            !dictionary.inheritedTypes.isEmpty {
+            return []
+        }
+        let substructureOffsets = dictionary.substructure.flatMap {
+            missingDocOffsets(in: $0, acls: acls)
+        }
+        guard (dictionary.kind).flatMap(SwiftDeclarationKind.init) != nil,
+            let offset = dictionary.offset,
+            let accessibility = dictionary.accessibility,
+            let acl = AccessControlLevel(identifier: accessibility),
+            acls.contains(acl) else {
+                return substructureOffsets
+        }
+        if dictionary.docLength != nil {
+            return substructureOffsets
+        }
+        return substructureOffsets + [(offset, acl)]
+    }
+}
+
+public struct MissingDocsRule: OptInRule {
+    public init(configuration: Any) throws {
+        guard let dict = configuration as? [String: Any] else {
```

Comment:
you should create a struct confirming to `RuleConfiguration` instead of parsing it here